### ISSUE TITLE
Feature: `injectable-should-be-provided` should also check `@Resolver()` classes

### ISSUE THREAD

**Issue by nkovacic** [darraghoriordan/eslint-plugin-nestjs-typed#463]
## Summary

`injectable-should-be-provided` currently checks `@Injectable()` and `@Controller()`. It doesn't check `@Resolver()`, so in a code-first GraphQL project a resolver that's missing from every module's `providers` array passes lint silently.

Resolvers are registered in `providers` — the same array the rule already checks for injectables — so this looks like it fits the existing shape rather than needing a new rule.

## Current behaviour

[`injectableShouldBeProvided.ts#L208-L225`](https://github.com/darraghoriordan/eslint-plugin-nestjs-typed/blob/main/src/rules/injectablesShouldBeProvided/injectableShouldBeProvided.ts#L208-L225) (v7.3.4):

```ts
ClassDeclaration(node: TSESTree.ClassDeclaration): void {
    checkNode(node, "Injectable", "providers",   "injectableInModule",  context, nestModuleMap);
    checkNode(node, "Controller", "controllers", "controllersInModule", context, nestModuleMap);
},
```

`@Resolver()` isn't matched, so resolver classes are never considered.

## Why it matters

The failure is silent in a way the `@Injectable()` case usually isn't. An unprovided service normally surfaces as a DI resolution error the first time something injects it. An unregistered resolver just... isn't in the schema. No error at boot, no error at runtime — the fields it declares are simply absent, and you find out when a client 400s on an unknown field.

We hit this on a NestJS code-first codebase with 114 resolvers. Four were absent from the module wiring we generate our SDL from, and one of them was a `@Mutation` that had been undocumented in our published `schema.graphql` for an unknown length of time. Nothing in lint, typecheck, build, or CI flagged it.

## Reproduction

```ts
// cat.resolver.ts
import { Query, Resolver } from "@nestjs/graphql";

@Resolver(() => Cat)
export class CatResolver {
    @Query(() => Cat)
    cat(): Cat {
        return new Cat();
    }
}
```

```ts
// cat.module.ts — CatResolver deliberately omitted
import { Module } from "@nestjs/common";

@Module({
    providers: [CatService],
})
export class CatModule {}
```

**Expected:** an error along the lines of the existing `injectableInModule` message.
**Actual:** no error.

## Suggested change

A third `checkNode` call against the same `providers` array:

```ts
checkNode(node, "Resolver", "providers", "resolversInModule", context, nestModuleMap);
```

plus a `resolversInModule` entry alongside the existing two in [`messages`](https://github.com/darraghoriordan/eslint-plugin-nestjs-typed/blob/main/src/rules/injectablesShouldBeProvided/injectableShouldBeProvided.ts#L148-L151).

A couple of notes on why I think this needs no new option:

- `@Resolver()` is unambiguously a provider decorator — Nest itself stamps it with `ENTRY_PROVIDER_WATERMARK` in [`resolver.decorator.ts`](https://github.com/nestjs/graphql/blob/master/packages/graphql/lib/decorators/resolver.decorator.ts), the same watermark `@Injectable()` carries. A `@Resolver()` class outside every `providers` array is a bug in every case I can think of.
- Projects not using GraphQL have no `@Resolver()` classes, so the check is inert for them.
- `nodeHasDecoratorsNamed` matches by decorator name, so the call forms `@Resolver()`, `@Resolver(() => Cat)` and `@Resolver("Cat")` should all be covered without special handling.

Lower confidence, so treat as a separate question: `@Scalar()` classes are also registered in `providers`, and might be worth the same treatment. I checked and `@Scalar()` does *not* set `ENTRY_PROVIDER_WATERMARK` the way `@Resolver()` does, so the argument for it is weaker and I'd understand leaving it out.

## Environment

- `@darraghor/eslint-plugin-nestjs-typed` 7.3.4
- `@nestjs/graphql` 12.x, code-first with `autoSchemaFile`

Happy to open a PR if you'd take one.

**Comment by darraghoriordan** [comment]
yea best just to submit a pr, thank you 👍
